core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -212,6 +223,8 @@ protected void populateCommonFields(MapperService mapperService,
                 throw new IllegalArgumentException("Analyzer [" + luceneAnalyzer + "] doesn't exists");
             }
             suggestionContext.setAnalyzer(luceneAnalyzer);
+        } else {
+            suggestionContext.setAnalyzer(mapperService.searchAnalyzer());
         }
```

Comment:
In https://github.com/elastic/elasticsearch/blob/feature-suggest-refactoring/core/src/main/java/org/elasticsearch/search/suggest/phrase/PhraseSuggestParser.java#L166 there is a whole block that tries to set the analyzer based on the MappedFieldType. I'm wondering if this should go here as well, although I don't see anything similar in TermSuggestParser, and how that plays with CompletionSuggest. Maybe @areek can help here?